src/scripts/breeding/Breeding.ts

```diff
@@ -306,19 +311,25 @@ class Breeding implements Feature {
         if (hatched) {
             this._eggList[index](new Egg());
             this.moveEggs();
-            if (this._queueList().length) {
-                const nextEgg = this.createEgg(this._queueList.shift());
-                this.gainEgg(nextEgg);
-                if (!this._queueList().length) {
-                    Notifier.notify({
-                        message: 'Hatchery queue is empty.',
-                        type: NotificationConstants.NotificationOption.success,
-                        timeout: 1e4,
-                        sound: NotificationConstants.NotificationSound.Hatchery.empty_queue,
-                        setting: NotificationConstants.NotificationSetting.Hatchery.empty_queue,
-                    });
-                }
-            }
+            this.nextEggFromQueue();
+        }
+    }
+
+    private nextEggFromQueue(): void {
+        if (!this._queueList().length) {
+            return;
+        }
+
+        const nextEgg = this.createEgg(this._queueList.shift());
+        this.gainEgg(nextEgg);
+        if (!this._queueList().length) {
+            Notifier.notify({
+                message: 'Hatchery queue is empty.',
+                type: NotificationConstants.NotificationOption.success,
+                timeout: 1e4,
+                sound: NotificationConstants.NotificationSound.Hatchery.empty_queue,
+                setting: NotificationConstants.NotificationSetting.Hatchery.empty_queue,
+            });
```

Comment:
Shouldn't this be moved to the check above with the `return;`?

Reply:
That notification is displayed when the queue becomes empty following an egg being moved from the queue to an egg slot. If it were moved to the check above it would display every time there was an open egg slot with no queue and eggs gained progress or hatched.